### PR TITLE
assert array contents, indifferent of ordering

### LINKED ISSUE #3342: Fix order-dependency in test
The build failed non-deterministically with the following:

```
  1) Failure:
CommentTest#test_mentions [/home/travis/build/exercism/exercism.io/test/exercism/comment_test.rb:29]:
Expected: ["bob", "alice"]
  Actual: ["alice", "bob"]

```

### PR DESCRIPTION
**Pull request by definemethod**
I think this should fix #3342.

### CODE REVIEW

File: test/exercism/comment_test.rb [hunk 1]
```diff
@@ -26,7 +26,7 @@ def test_mentions
       ["Mention @alice and @bob and also @charlie", %w(alice bob), "ignore unknown users"],
     ].each do |body, mentions, _desc|
       usernames = User.where(id: Comment.new(body: body).mention_ids).map(&:username)
-      assert_equal usernames, mentions
+      assert_equal usernames.sort, mentions.sort
```
Comment (Insti) [inline test/exercism/comment_test.rb]: This is not your bug, but the arguments here are the wrong way around, it should be:
`assert_equal expected, actual`

The `sort`s are a good solution to the reported non-deterministic ordering issue.

Comment (mhelmetag) [inline test/exercism/comment_test.rb]: In general, I think this is fine. Another way to test this is just to make sure both arrays include the same values regardless of order (probably could be done a few ways but one way is to just minus the arrays and check that there's nothing left).

Comment (Insti) [inline test/exercism/comment_test.rb]: If you pre-sort the expected values, you only need to explicitly sort the result values.

Comment (mhelmetag) [inline test/exercism/comment_test.rb]: With the current sort solution, I think this is a good way to go about it and reduce the amount of sorts.

Comment (definemethod) [inline test/exercism/comment_test.rb]: I'm happy to presort the existing values if needed.

**Comment by definemethod** [comment]
The expected values were already sorted, and I went ahead and reordered the actual and expected values.

**Comment by kytrinyx** [comment]
✨ Thank you! ✨

**Pull request merged**